packages/amplify_auth_cognito/example/ios/amplify_auth_cognito_tests/MockAuthSession.swift

```diff
@@ -0,0 +1,104 @@
+/*
+* Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+import Amplify
+import AmplifyPlugins
+import AWSPluginsCore
+import AWSCore
+import AWSMobileClient
+
+class MockCredentials: AuthAWSCredentials {
```

Comment:
Martin Fowler [cites](https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs) Gerard's Meszaro's [xUnit Test Patterns](https://martinfowler.com/books/meszaros.html), where the following definitions are provided for different test doubles:
- **Dummy** objects are passed around but never actually used. Usually they are just used to fill parameter lists.
- **Fake** objects actually have working implementations, but usually take some shortcut which makes them not suitable for production (an in memory database is a good example).
- **Stubs** provide canned answers to calls made during the test, usually not responding at all to anything outside what's programmed in for the test.
- **Spies** are stubs that also record some information based on how they were called. One form of this might be an email service that records how many messages it was sent.
- **Mocks** are what we are talking about here: objects pre-programmed with expectations which form a specification of the calls they are expected to receive.

I believe that all of these `Mock*` classes you have, which are simplified implementations of `protocol`s, should be called `Fake*`s?

Reply:
Thanks @jamesonwilliams  - I made this change.